packages/data-stores/src/mutations/use-site-reset-mutation.ts

```diff
@@ -0,0 +1,36 @@
+import { UseMutationOptions, useMutation } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wpcomRequest from 'wpcom-proxy-request';
+
+interface APIResponse {
```

Comment:
I added some types and it's looking good. Any additional changes can be done in another PR.